src/os/bluestore/BlueFS.cc

```diff
@@ -2292,3 +2292,10 @@ int BlueFS::unlink(const string& dirname, const string& filename)
   _drop_link(file);
   return 0;
 }
+
+bool BlueFS::wal_is_rotational()
+{
+  if (!bdev[BDEV_WAL] || bdev[BDEV_WAL]->is_rotational())
```

Comment:
wouldn't the all-ssd case satisfy the first condition? seems like we'd need to check the main device as well in that case